tests/foreman/ui_airgun/test_computeresource.py

```diff
@@ -0,0 +1,156 @@
+    with session:
+        session.computeresource.create({
+            'name': name,
+            'description': gen_string('alpha'),
+            'provider': 'VMware',
+            'provider_content.vcenter': vmware_vcenter,
+            'provider_content.user': vmware_user,
+            'provider_content.password': vmware_password,
+        })
+        assert session.computeresource.search(name)[0]['Name'] == name
+
+
+@parametrize('name', **valid_data_list('ui'))
+def test_positive_create_rhv(session, name):
+    rhev_url = settings.rhev.hostname
+    username = settings.rhev.username
+    password = settings.rhev.password
+    with session:
+        session.computeresource.create({
+            'name': name,
+            'description': gen_string('alpha'),
+            'provider': 'RHV',
+            'provider_content.url': rhev_url,
+            'provider_content.user': username,
+            'provider_content.password': password,
+        })
+        assert session.computeresource.search(name)[0]['Name'] == name
+
+
+def test_positive_rename(session):
+    name = gen_string('alpha')
+    ak_name = gen_string('alpha')
+    docker_url = settings.docker.external_url
+    with session:
+        session.computeresource.create({
+            'name': name,
+            'description': gen_string('alpha'),
+            'provider': 'Docker',
+            'provider_content.url': docker_url,
+        })
+        session.computeresource.edit(name, {
+            'name': ak_name,
+        })
+        assert session.computeresource.search(ak_name)[0]['Name'] == ak_name
+
+
+def test_positive_delete(session):
+    name = gen_string('alpha')
+    docker_url = settings.docker.external_url
+    with session:
+        session.computeresource.create({
+            'name': name,
+            'description': gen_string('alpha'),
+            'provider': 'Docker',
+            'provider_content.url': docker_url,
+        })
+        session.computeresource.delete(name)
+        assert not session.computeresource.search(name)
+
+
+@parametrize('name', valid_data_list())
+def test_positive_v3_wui_can_add_resource(session, name, version=3):
+    """Create new RHEV Compute Resource using APIv3 and autoloaded cert"""
+    rhev_url = settings.rhev.hostname
+    username = settings.rhev.username
+    password = settings.rhev.password
+    with session:
+        session.computeresource.create({
+            'name': name,
+            'provider': 'oVirt',
+            'url': rhev_url,
+            'user': username,
+            'password': password,
+            'api4': version==4,
+        })
+        assert session.computeresource.search(name) == name
+        assert session.computeresource.read(name)['api4'] == (version==4)
+
+@parametrize('name', valid_data_list())
+def test_positive_v4_wui_can_add_resource(session, name):
+    """Create new RHEV Compute Resource using APIv4 and autoloaded cert"""
+    test_positive_v3_wui_can_add_resource(session, name, version=4)
```

Comment:
we never use such style in the tests, please whether copy/paste entire test logic and change necessary arguments or use parametrize

Reply:
TIL it's possible to call one test from the inside of another 😄 
If the only thing that changes is `version` - i'd prefer `parametrize` over copy/paste